x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/CcrLicenseChecker.java

```diff
@@ -244,6 +244,11 @@ public void fetchLeaderHistoryUUIDs(
         String leaderIndex = leaderIndexMetaData.getIndex().getName();
         CheckedConsumer<IndicesStatsResponse, Exception> indicesStatsHandler = indicesStatsResponse -> {
             IndexStats indexStats = indicesStatsResponse.getIndices().get(leaderIndex);
+            if (indexStats == null) {
+                onFailure.accept(new IllegalArgumentException("no index shards available, is the leader index red?"));
```

Comment:
How about just saying - `no index stats available for the leader index` ? The leader index can be red when missing just one primary and you'd still have stats. Also maybe something else caused this and we're just misleading people? I prefer to keep this simple.